content/show/coder-radio/_index.md

```diff
@@ -14,6 +14,11 @@ header_image = "/images/shows/coder-radio.png"
 
 podverse_podcast_id = "ZXd_1Ojd9"
 
+[live]
+  type = "weekly"
+  day = 1
+  hour = 9
```

Comment:
The Coder Radio card seems to be ahead of the Google Calendar by about 3 hours.

![image](https://user-images.githubusercontent.com/34293941/217002116-60a898ec-66f8-490c-9720-5b28f4747067.png)

Reply:
I _think_ this is because today's time is not the regular time.  It should match if you look at the time for next week.  I can ask around too though.

But I guess it does bring up a non-technical question, is there a way we can make the reason these don't match more clear?  Not something we can do in the website, but could the calendar items add "(Special Time)" to the title, or something similar? 

(And I will take a swing at the other comments later, thanks!)

Reply:
[Chris confirmed in matrix](https://matrix.to/#/!teHukPkJDFppjPjGQg:matrix.org/$Z0zqsiYcqdoFtIuh2iV-PklWcu5wPhm30XnfETMcVAE?via=matrix.org&via=jupiterbroadcasting.com&via=linuxdelta.com) that today's time is not the usual time.

Reply:
Looks like I should've looked farther ahead! I think we the calendar should have some kind of note if possible to specify that the time is different from normal. I haven't used Google Calendar so I am not sure if that is an option. We can ask Chris about that

Reply:
Another crazy thought... Could we use the Google Calendar API to get the time of the next event? Maybe outside the scope of this PR (unless you really want to tackle it). But that could be a good next step on this.

Reply:
I wouldn't want to drag this out too much if the current version seems useful as a first iteration.

But separately, I wouldn't mind looking into using the Google Calendar API.  I'd guess the best way would be to call it during the Hugo build, instead of client side JS, so we don't have to worry about API keys being public?  If we could copy the next two months of showtimes into the static files, we could include "Next Showtime: ..." in each card.

For now, what if we put a header between the cards and the calendar that says something like "Upcoming Times - Includes Special Times"

Reply:
We are looking into how to handle the contact form with its API as well. I had proposed a separate backend just handle API requests like those. There was some discussion about it in #232

Reply:
> so we don't have to worry about API keys being public?

We can look into settings to protect and scope the Google Calendar API key properly. Such as only allowing access to the JB Calendar from jupiterbroadcasting.com, and nothing else from Google Calendar. I haven't worked with the Google Calendar API, but I've seen other APIs allow these kinds of settings, so it should work with Google Calendar as well. Quickly glancing at the API, it seems to be possible with https://cloud.google.com/docs/authentication/api-keys#api_key_restrictions.

I see https://irc.twit.tv/ is using a solution based on the Google Calendar API, we might be able to build something similar. You can search for "calendar" in the Network tab of your browser and see what Google API they are using. I'm hoping they've done their research on not expoing their API key.

If we have an API key, this might just be a separate AJAX request to a JSON API, similar to what TWiT is using.

> I wouldn't want to drag this out too much if the current version seems useful as a first iteration.

I agree, but I think if we change the calendar, we should make sure it's correct and not too confusing. I think the design of the cards looks great, and it's definitely a great addition. Just hard-coding the events might make it more confusing. A better solution could be using the Google Calendar API, in my opinion.

Reply:
I did get excited to have something merged here, but makes sense to back up and reassess.  😅

Are we still talking about adding "Next Showtime" to the cards?  And potentially using the Google Calendar API to determine that?  

We'd still need to indicate when the "Next Showtime" is not a regular show time.  We could maybe do that by calculating when the next regular time _should_ be, and if they don't match, we know it's special.

Reply:
> We could maybe do that by calculating when the next regular time should be, and if they don't match, we know it's special.

That could work. Maybe looking at the last 3-4 shows and determining a pattern could be an idea.

We should also take permanent changes in the live calendar into account, like how LUP was moved from Tuesday to Sunday.

I also noticed that some event in the calendar have a different name to make it clear it was a different live time.

![image](https://user-images.githubusercontent.com/35609464/218272278-93dbe94d-4874-4de1-b5ab-39248486a294.png)

Maybe this will make it more difficult to automatically detect which show is live? We could detect any show with "Coder Radio" in the title, but what about false positives like "No Coder Radio this week"? Have we had this before? 😁

Or maybe it's good enough to have a system that works most of the time? Google has changed the API before, and there will be changes in the future, so there might always be unforeseen breakages.

Reply:
> "No Coder Radio this week"

Nevermind I give up.

Just kidding, but really my goal here was to have somewhere on the website to find general live schedules for each show.  We immediately hit a corner case where it was confusing, so it became more difficult to proceed.  If I have time I'll check out the Google Calendar API.

> Maybe looking at the last 3-4 shows and determining a pattern could be an idea.

I'd hope we can be more precise, either by looking at the recurrence of the Google Calendar events, or having something like I have in the `live` property of shows in this PR.